src/main/java/com/ookawara/animeapi/service/AnimeServiceImpl.java

```diff
@@ -23,7 +24,13 @@ public List<Anime> findAll(){
 
     @Override
     public Optional<Anime> findById(int id){
-        return animeMapper.findById(id);
+        Optional<Anime> anime = this.animeMapper.findById(id);
+        if (anime.isPresent()){
+            return Optional.of(anime.get());
+        } else {
+            throw new AnimeNotFoundException("ID:" + id + "のデータは存在しません。");
+        }
+
```

Comment:
OptionalはorElseThrowを使うことで、下記のように簡潔に書くことができます！
```suggestion    
    public Anime findById(int id){
        return this.animeMapper.findById(id)
                    .orElseThrow(() -> new AnimeNotFoundException("ID:" + id + "のデータは存在しません。"));
    }
```

Reply:
Controller側でList<Anime>をレスポンスするようになっていましたが、idに紐づくanimeが複数になることはないと思うので、Listじゃない方が良さそうですね！

Reply:
> updateの処理と同様にorElseThrowを使ってあげて、返り値の方をAnimeにしてあげたら良いのかなと思ったのですが、いかがでしょうか？
これって、自分の理解が追いついていないので確認したいのですが、updateと同じ書き方同じ書き方にしてみては？ということで理解合っていますか？
今回は自分の理解を深める意味を込めて、if(){ }else{ throw ...} という形でも処理することができるということを残しておきたくてあえてこの表記で書いてみました。

Reply:
> Controller側でListをレスポンスするようになっていましたが、idに紐づくanimeが複数になることはないと思うので、Listじゃない方が良さそうですね！

こちらは画像の赤丸のことで理解合っていますか？
<img width="1280" alt="スクリーンショット 2023-09-01 1 39 19" src="https://github.com/kawara777/Chapter9/assets/138858245/7d4f75de-a41d-4803-afc9-fe3bb2a817f5">

Reply:
> これって、自分の理解が追いついていないので確認したいのですが、updateと同じ書き方にしてみては？ということで理解合っていますか？
今回は自分の理解を深める意味を込めて、if(){ }else{ throw ...} という形でも処理することができるということを残しておきたくてあえてこの表記で書いてみました。

おっしゃる通りupdateと同じ書き方の方が簡潔なのでという意味でした！！
ただ、おっしゃるように学習のためということですので、このままでも大丈夫ですよ！！

その場合、
```
return Optional.of(anime.get());
```
としているところについては、
```
return anime.get();
```
の方が良いと思います。
すでにif文でanimeがnull出ないことが確定しているので、あえてまたOptional型でreturnする必要がないからです。

Reply:
> こちらは画像の赤丸のことで理解合っていますか？

はい、変数名はanimeListですが、実際にreturnされている値はListではないですよね！
このControllerのfindByIdの返却値もList<Anime>ではなく、Animeで良いのではないかと思います！